server/tests/batch_comparison/test_sample_extra_batches_by_counting_group.py

```diff
@@ -0,0 +1,339 @@
+    choice_ids = [choice["id"] for choice in contests[0]["choices"]]
+
+    batch_results = {
+        j1_batches[0]["id"]: {
+            choice_ids[0]: 400,
+            choice_ids[1]: 50,
+            choice_ids[2]: 40,
+        },
+        j1_batches[1]["id"]: {
+            choice_ids[0]: 100,
+            choice_ids[1]: 50,
+            choice_ids[2]: 40,
+        },
+        j1_batches[2]["id"]: {
+            choice_ids[0]: 100,
+            choice_ids[1]: 50,
+            choice_ids[2]: 40,
+        },
+        # The extra batch
+        j1_batches[3]["id"]: {choice_ids[0]: 0, choice_ids[1]: 0, choice_ids[2]: 0,},
+    }
+
+    for batch_id, results in batch_results.items():
+        set_logged_in_user(
+            client, UserType.JURISDICTION_ADMIN, default_ja_email(election_id)
+        )
+        rv = put_json(
+            client,
+            f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[0]}/round/{round_1_id}/batches/{batch_id}/results",
+            [{"name": "Tally Sheet #1", "results": results}],
+        )
+        assert_ok(rv)
+
+    # Finalize the results
+    rv = post_json(
+        client,
+        f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[0]}/round/{round_1_id}/batches/finalize",
+    )
+    assert_ok(rv)
+
+    # Now do the second jurisdiction
+    batch_results = {
+        j2_batches[0]["id"]: {
+            choice_ids[0]: 100,
+            choice_ids[1]: 100,
+            choice_ids[2]: 40,
+        },
+        # The extra batch
+        j2_batches[1]["id"]: {
+            choice_ids[0]: 1,
+            choice_ids[1]: 200,
+            choice_ids[2]: 200,
+        },
+    }
+
+    for batch_id, results in batch_results.items():
+        set_logged_in_user(
+            client, UserType.JURISDICTION_ADMIN, default_ja_email(election_id)
+        )
+        rv = put_json(
+            client,
+            f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[1]}/round/{round_1_id}/batches/{batch_id}/results",
+            [{"name": "Tally Sheet #1", "results": results}],
+        )
+        assert_ok(rv)
+
+    rv = post_json(
+        client,
+        f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[1]}/round/{round_1_id}/batches/finalize",
+    )
+    assert_ok(rv)
+
+    # Check the audit report
+    set_logged_in_user(client, UserType.AUDIT_ADMIN, DEFAULT_AA_EMAIL)
+    rv = client.get(f"/api/election/{election_id}/report")
+    assert_match_report(rv.data, snapshot)
+
+    # The audit results should be the same as the audit results without the
+    # extra sampled batches
+    report = scrub_datetime(rv.data.decode("utf-8"))
+    # Generated by turning off the feature flag
+    expected_audit_results_line = "1,Contest 1,Targeted,6,No,0.1857414858,DATETIME,DATETIME,candidate 1: 700; candidate 2: 250; candidate 3: 160"
+    assert expected_audit_results_line in report
+
+
+def test_sample_extra_batches_with_no_extra_batches_to_sample(
```

Comment:
Only think that sticks out to me: could we include a pathological test case for jxns that only have a batch of one type? I.e. it's not unlikely that in small counties there will only be one HMPB batch maybe? I don't think that will cause any issues, but want to make sure we're testing for it.